implementation/src/main/java/io/smallrye/config/ProfileConfigSourceInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package io.smallrye.config;
+
+import java.util.Optional;
+
+import javax.annotation.Priority;
+
+@Priority(600)
+public class ProfileConfigSourceInterceptor implements ConfigSourceInterceptor {
+    public static final String SMALLRYE_PROFILE = "smallrye.config.profile";
+
+    private final String profile;
+
+    public ProfileConfigSourceInterceptor(final String profile) {
+        this.profile = profile;
+    }
+
+    public ProfileConfigSourceInterceptor(final ConfigSourceInterceptorContext context) {
+        this(context, SMALLRYE_PROFILE);
+    }
+
+    public ProfileConfigSourceInterceptor(
+            final ConfigSourceInterceptorContext context,
+            final String profileConfigName) {
+        this.profile = Optional.ofNullable(context.proceed(profileConfigName)).map(ConfigValue::getValue).orElse(null);
+    }
+
+    @Override
+    public ConfigValue getValue(final ConfigSourceInterceptorContext context, final String name) {
+        if (profile != null) {
+            final ConfigValue profileValue = context.proceed("%" + profile + "." + name);
+            if (profileValue != null) {
+                final ConfigValue originalValue = context.proceed(name);
+                if (originalValue != null) {
+                    return originalValue.getConfigSourceOrdinal() > profileValue.getConfigSourceOrdinal() ? originalValue
```

Comment:
One more tiny detail... if the ordinals are equal, we then disambiguate based on the name of the config source as per https://github.com/smallrye/smallrye-config/blob/master/implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java#L56-L65 so we probably want a simlar check here.

Reply:
Ok!